src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -22,54 +19,40 @@ object SwaggerService {
     idiomatic (but not unheard of) from().
    */
   def from(conf: Config)(implicit actorRefFactory: ActorRefFactory): SwaggerService = {
-    val swaggerConfig = conf.getConfig("swagger")
-    new SwaggerService(
-      swaggerConfig.getString("apiVersion"),
-      swaggerConfig.getString("baseUrl"),
-      swaggerConfig.getString("apiDocs"),
-      swaggerConfig.getString("swaggerVersion"),
-      Vector(typeOf[CromwellApiService]),
-      Option(new ApiInfo(
-        swaggerConfig.getString("info"),
-        swaggerConfig.getString("description"),
-        swaggerConfig.getString("termsOfServiceUrl"),
-        swaggerConfig.getString("contact"),
-        swaggerConfig.getString("license"),
-        swaggerConfig.getString("licenseUrl"))
-      ))
+    new SwaggerService(conf.getConfig("swagger"))
   }
 }
 
-class SwaggerService(override val apiVersion: String,
-                     override val baseUrl: String,
-                     override val docsPath: String,
-                     override val swaggerVersion: String,
-                     override val apiTypes: Seq[Type],
-                     override val apiInfo: Option[ApiInfo])
-                    (implicit val actorRefFactory: ActorRefFactory) extends SwaggerHttpService
+class SwaggerService(override val swaggerConfig: Config)
+                    (implicit val actorRefFactory: ActorRefFactory)
+  extends SwaggerConfigHttpService {
+  override def apiTypes = Vector(typeOf[CromwellApiService])
+}
 
 object CromwellApiServiceActor {
-  def props(workflowManagerActorRef : ActorRef, swaggerService: SwaggerService): Props = {
+  def props(workflowManagerActorRef: ActorRef, swaggerService: SwaggerService): Props = {
     Props(new CromwellApiServiceActor(workflowManagerActorRef, swaggerService))
   }
 }
 
-class CromwellApiServiceActor(val workflowManager : ActorRef, swaggerService: SwaggerService) extends Actor with CromwellApiService {
+class CromwellApiServiceActor(val workflowManager: ActorRef, swaggerService: SwaggerService) extends Actor with CromwellApiService {
   implicit def executionContext = actorRefFactory.dispatcher
   def actorRefFactory = context
 
-  def possibleRoutes = workflowRoutes ~ swaggerService.routes ~
-    get {
-      pathSingleSlash {
-        getFromResource("swagger/index.html")
-      } ~ getFromResourceDirectory("swagger/") ~ getFromResourceDirectory("META-INF/resources/webjars/swagger-ui/2.0.24/")
-    }
+  def possibleRoutes = workflowRoutes ~ swaggerService.routes ~ swaggerService.uiRoutes
 
   def receive = runRoute(possibleRoutes)
 }
 
+object CromwellApiService {
+  /**
+   * Comma separated list of optional endpoint versions. The last value is the latest version.
+   */
```

Comment:
Unless I'm being dense, as this value is used in an annotation, and must stay a static / non-generated string? Updated this variable name and comment to hopefully make it clearer where it's used below.